perfkitbenchmarker/virtual_machine.py

```diff
@@ -235,6 +235,17 @@ def AddMetadata(self, **kwargs):
     """
     pass
 
+  def GetMachineTypeDict(self):
```

Comment:
Yes, the idea here is that sometimes a "machine type" isn't defined by a single string like "n1-standard-8". When using a custom GCE VM, we could instead have, for example, {'cpus': 2, 'memory_mb': 4.0, 'preemptible': True} that collectively defines the type of machine that was provisioned. The reason I used "MachineType" is that this the equivalent of what our previous machine_type field did. "Properties" might imply that there could be other info in here, like number of disks or OS image, whereas those are separate properties that aren't related to our current understanding of a "machine type".